### ISSUE TITLE
Parser: Nested array destructuring not parsing correctly

### ISSUE THREAD

**Issue by sukus21** [grafana/sobek#48]
This was also an issue in Goja: https://github.com/dop251/goja/issues/614

Nested array destructuring is not working as intended, here is a concrete example I've run into:

```
// Code from (what looks like) a matrix math function
var [
    [o, a = o, l = a],
    [h, /* issue occurs here */ u = h, c = u] = [0, 0, 0],
    [d, f = d, p = f] = [h, u, c],
    [g, m = g, y = m] = [o, a, l],
] = [/* array values */];
```

**Comment by mstoykov** [comment]
Thank you for reporting this @sukus21.

I am aware, but waiting on a different issue to potentially be fixed as it is more directly interesting as at least I have never hit the issue you showcase. 

So we will probably merge the fix from goja along with the fix for the other issue. 

Do you currently use sobek and this is an immediate issue for you?

**Comment by sukus21** [comment]
I am not currently using sobek, but I am considering switching from Goja. I am only using it for a hobby project, so it is not urgent in the slightest

**Comment by mstoykov** [comment]
Fixed in 165518de5203d54ae1455cd425d6ddd465d43105
